drivers/network/netio/netio.c

```diff
@@ -0,0 +1,882 @@
+			case SO_WSK_EVENT_CALLBACK:
+				if (InputBuffer == NULL) {
+					DbgPrint("WskControlSocket: Need an InputBuffer for this operation\n");
+					status = STATUS_INVALID_PARAMETER;
+					break;
+				}
+				if (InputSize < sizeof(WSK_EVENT_CALLBACK_CONTROL)) {
+					DbgPrint("WskControlSocket: InputBuffer too small for this operation\n");
+					status = STATUS_INVALID_PARAMETER;
+					break;
+				}
+				WSK_EVENT_CALLBACK_CONTROL *c = (WSK_EVENT_CALLBACK_CONTROL*)InputBuffer;
+
+					/* TODO: compare this bytewise */
+/*				if (c->NpiId != &NPI_WSK_INTERFACE_ID) {
+					DbgPrint("WskControlSocket: Only NPI_WSK_INTERFACE_ID supported, sorry.\n");
+					status = STATUS_INVALID_PARAMETER;
+					break;
+				} */
+				s->CallbackMask = c->EventMask;
+					/* TODO: and start listening here? */
+
+				status = STATUS_SUCCESS;
+				break;
+
+			default:
+				DbgPrint("WskControlSocket: ControlCode %d Not implemented.\n", ControlCode);
+			}
+			break;
+		default:
+			DbgPrint("WskControlSocket: Level %d Not implemented.\n", Level);
+		}
+		break;
+
+	case WskGetOption:
+	case WskIoctl:
+	default:
+		DbgPrint("WskControlSocket: Option %d Not implemented.\n", RequestType);
+	}
+	if (Irp != NULL) {
+		Irp->IoStatus.Status = status;
+/* ??		Irp->IoStatus.Information = Irp->IoStatus.Information; */
+	}
+	return status;
+}
+
+static NTSTATUS WSKAPI WskCloseSocket(
+    _In_ PWSK_SOCKET Socket,
+    _Inout_ PIRP Irp)
+{
+	struct _WSK_SOCKET_INTERNAL *s = (struct _WSK_SOCKET_INTERNAL*) Socket;
+
+	SocketPut(s);
+	/* TODO: DummyCallDriver and IoComplete */
+	return STATUS_SUCCESS;
+}
+
+static struct _TRANSPORT_ADDRESS *TdiTransportAddressFromSocketAddress(PSOCKADDR SocketAddress)
+{
+	struct _TRANSPORT_ADDRESS *ta;
+
+	ta = ExAllocatePoolWithTag(NonPagedPool, sizeof(*ta)+sizeof(struct sockaddr) , 'ADDR');
+	if (ta == NULL) {
+		DbgPrint("TdiTransportAddressFromSocketAddress: Out of memory\n");
+		return NULL;
+	}
+
+//	memset(ta, 0, sizeof(*ta)+sizeof(struct sockaddr));
+	ta->TAAddressCount = 1;
+	ta->Address[0].AddressLength = sizeof(SocketAddress->sa_data);
+	ta->Address[0].AddressType = SocketAddress->sa_family;
+	memcpy(&ta->Address[0].Address[0], &SocketAddress->sa_data, ta->Address[0].AddressLength);
+
+	return ta;
+}
+
+static PTDI_CONNECTION_INFORMATION TdiConnectionInfoFromSocketAddress(PSOCKADDR SocketAddress)
+{
+	PTRANSPORT_ADDRESS TargetAddress;
+	PTDI_CONNECTION_INFORMATION ConnectionInformation = NULL;
+	NTSTATUS status;
+
+	TargetAddress = TdiTransportAddressFromSocketAddress(SocketAddress);
+	if (TargetAddress == NULL)
+		return NULL;
+
+	status = TdiBuildConnectionInfo(&ConnectionInformation, TargetAddress);
+	ExFreePoolWithTag(TargetAddress, 'ADDR');
+
+	if (!NT_SUCCESS(status)) {
+		if (ConnectionInformation != NULL) {
+			ExFreePoolWithTag(ConnectionInformation, TAG_AFD_TDI_CONNECTION_INFORMATION);
+		}
+		return NULL;
+	}
+	return ConnectionInformation;
+}
+
+
+static NTSTATUS WSKAPI WskBind (
```

Comment:
```suggestion
static NTSTATUS WSKAPI WskBind(
```
Here and in other locations.

Reply:
Hi I ran 

     indent -o %2 -gnu -bli0 -hnl -nut -i4 -l110 -bls -ncs -npcs -nprs -nsaf -nsob -lp -cli4 -cbi0 %1

and it looks like it fixed these.